src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -181,6 +181,9 @@ private static class StaticWatcherRunLoopManager
 
                 private static readonly object s_lockObject = new object();
 
+                private static readonly AutoResetEvent s_runLoopStartedEvent = new AutoResetEvent(false);
+                private static readonly AutoResetEvent s_runLoopEndedEvent = new AutoResetEvent(false);
```

Comment:
I'm not clear on why we need these AutoResetEvents or the spin loop you added.  Is that all to address my concern about the race condition on the IntPtr?  If so, why wouldn't a better fix have been to just not set it to zero in the thread worker, since it was already being set to zero by the last watcher to stop?